### PR TITLE
issue/2991-freshly-pressed-v1.2

### LINKED ISSUE #2991: Reader: use v1.2 Freshly Pressed endpoint
In the [reader discover branch](https://github.com/wordpress-mobile/WordPress-Android/pull/2978) we switched to v1.2 of the REST client when requesting posts, but a [hack](https://github.com/wordpress-mobile/WordPress-Android/commit/5858a3e1de4b4d220b9a2ab33e20e51293a51f86#diff-b223b02ca0f90eda2bda3d2a2398025bR202) was added to use v1.1 for Freshly Pressed since there currently isn't a v1.2 FP endpoint yet.

This issue is a reminder to switch to the v1.2 FP endpoint when it becomes available.

### PR DESCRIPTION
**Pull request by nbradbury**
TL;DR: If this PR is bad, then Freshly Pressed will fail to update in the reader. If FP does update, it's good.

Fix #2991 - This simple PR removes [this hack](https://github.com/wordpress-mobile/WordPress-Android/commit/5858a3e1de4b4d220b9a2ab33e20e51293a51f86#diff-b223b02ca0f90eda2bda3d2a2398025bR202) which was added in a previous release to work around an issue in the `read/menu` endpoint.

That endpoint returns a list of the endpoints the app should call in order to update posts in the reader, and at the time it was returning v1.2 endpoints for everything except Freshly Pressed, which required calling a v1.1 endpoint.

This has since been corrected on the server side, so the hack is no longer required.

### CODE REVIEW

**Comment by maxme** [comment]
It's not related to this PR, but I noticed a weird thing with FP posts dates: https://cloudup.com/ceV62SnKlxq - I guess FP published date != post published date - so probably not fixable.

**Comment by nbradbury** [comment]
Yes, the published date is actually the FP date.
